drivers/dma/dma_mcux_fifo.c

```diff
@@ -0,0 +1,117 @@
+#include <zephyr/drivers/dma/dma_mcux_fifo.h>
+#include "fsl_edma.h"
+#include "fsl_dmamux.h"
+#include "math.h"
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <errno.h>
+#include <soc.h>
+#include <zephyr/sys/atomic.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/cache.h>
+#include <MIMXRT1176_cm7.h>
+#include <errno.h>
+#include <fsl_common.h>
+
+#define DT_DRV_COMPAT nxp_mcux_fifo
+
+char *get_values(const struct device *dev, uint16_t count)
+{
+	return NULL;
+}
+
+bool set_callback(const struct device *dev, callback_t callback_fun)
+{
+	dma_mcux_fifo_data_t *data = (dma_mcux_fifo_data_t *)dev->data;
+	data->fifo_chunk_ready = callback_fun;
+	return true;
+}
+
+int enable_fifo(const struct device *dev, bool enable)
+{
+	cache_data_disable();
+	dma_mcux_fifo_config_t *config = (dma_mcux_fifo_config_t *)dev->config;
+	DMA_Type *edma = config->dma_base;
+	DMAMUX_Type *mux = config->dmamux_base;
+	if (enable) {
+		uint32_t interrupts =
+			DMA_CSR_INTMAJOR_MASK | DMA_CSR_INTHALF_MASK | kEDMA_ErrorInterruptEnable;
+		dma_mcux_fifo_data_t *data = (dma_mcux_fifo_data_t *)dev->data;
+		EDMA_SetModulo(edma, config->dma_channel1, kEDMA_Modulo2bytes, kEDMA_Modulo64bytes);
+		void *callbackptr = data->fifo_chunk_ready;
+		EDMA_EnableChannelInterrupts(edma, config->dma_channel1, interrupts);
+		EDMA_EnableChannelRequest(edma, config->dma_channel1);
+		irq_connect_dynamic(config->dma_channel1 % 16, 5, callbackptr, dev, 0);
+		irq_connect_dynamic(DMA_ERROR_IRQn, 5, callbackptr, dev, 0);
+		irq_enable(config->dma_channel1 % 16);
+		irq_enable(DMA_ERROR_IRQn);
+		EDMA_EnableAutoStopRequest(edma,config->dma_channel1,false);
+		DMAMUX_SetSource(mux, config->dma_channel1, (uint32_t)config->dma_trigger);
+		DMAMUX_EnableChannel(mux, config->dma_channel1);
+	} else {
+		DMAMUX_DisableChannel(mux, config->dma_channel1);
+	}
+	cache_data_enable();
+	return true;
+}
+
+static int dma_mcux_fifo_init(const struct device *dev)
+{
+	cache_data_disable();
+	dma_mcux_fifo_data_t *data = (dma_mcux_fifo_data_t *)dev->data;
+	dma_mcux_fifo_config_t *config = (dma_mcux_fifo_config_t *)dev->config;
+	data->fifo_data = k_malloc(pow(2, config->fifo_size_log2));
```

Comment:
I think `pow` returns a float or double. I suggest to either cast the result explicitly back to `size_t` or just use a bit shift.

 ```suggestion
	data->fifo_data = k_malloc((size_t)pow(2, config->fifo_size_log2));
```

```suggestion
	data->fifo_data = k_malloc( 1u << fifo_size_log2);
```

Reply:
jup, makes sense